src/jp2image.cpp

```diff
@@ -499,8 +499,8 @@ namespace Exiv2
                             subBox.length = getLong((byte*)&subBox.length, bigEndian);
                             subBox.type   = getLong((byte*)&subBox.type, bigEndian);
 
-                            // subBox.length makes no sense if it is larger than the rest of the file
-                            if (subBox.length > io_->size() - io_->tell()) {
+                            // subBox.length makes no sense if it is larger than the rest of the file || 0
+                            if (subBox.length == 0 || subBox.length > io_->size() - io_->tell()) {
```

Comment:
This check is not exhaustive: imagine `subBox.length` being `sizeof(box) - 1`, then in line 506 you initialize `data` with `-1` and the whole thing blows up.

I would suggest that instead of throwing an exception here, we simply limit the size of `data` to sane values.

Reply:
```
                            // subBox.length makes no sense if it is larger than the rest of the file
                            if (subBox.length > io_->size() - io_->tell()) {
                                throw Error(kerCorruptedMetadata);
                            }
                            // subBox.length must be > sizeof(subBox)
                            if (subBox.length <= sizeof(subBox) ) {
                                throw Error(kerCorruptedMetadata);
                            }
```

There's an issue with `makeSlice(data,0,30)` on line 512.  https://github.com/Exiv2/exiv2/issues/744

I've just noticed there's probably an issue with '40' on makeSlide(rawData,0,40) on line 560.  This can be remedied with code similar to my suggestion for #744.

We'll need an additional test `box.length > sizeof(uuid) + sizeof(box)` about line 554 before allocating rawData.

Reply:
@D4N I slightly changed the check for `subBox.length` as you suggested and added the python test.

@clanmills Let's analyse the other issue in a different branch. I do not want to mix concepts here. This PR is just about fixing #742